modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/web/internal/object/entries/frontend/data/set/test/RelatedModelFDSActionProviderTest.java

```diff
@@ -0,0 +1,201 @@
+			TestPropsValues.getUserId(), 0,
+			objectDefinition.getObjectDefinitionId(), Collections.emptyMap(),
+			ServiceContextTestUtil.getServiceContext());
+
+		Role role = RoleTestUtil.addRole(RoleConstants.TYPE_REGULAR);
+
+		_resourcePermissionLocalService.setResourcePermissions(
+			TestPropsValues.getCompanyId(), objectDefinition.getClassName(),
+			ResourceConstants.SCOPE_COMPANY,
+			String.valueOf(TestPropsValues.getCompanyId()), role.getRoleId(),
+			new String[] {ActionKeys.UPDATE});
+
+		User user = UserTestUtil.addUser();
+
+		_userLocalService.addRoleUser(role.getRoleId(), user);
+
+		PermissionThreadLocal.setPermissionChecker(
+			PermissionCheckerFactoryUtil.create(user));
+
+		PrincipalThreadLocal.setName(user.getUserId());
+
+		List<DropdownItem> dropdownItems = _fdsActionProvider.getDropdownItems(
+			TestPropsValues.getGroupId(),
+			_getMockHttpServletRequest(objectEntry.getObjectEntryId()),
+			new RelatedModel(
+				objectDefinition.getClassName(), objectEntry.getObjectEntryId(),
+				objectEntry.getTitleValue(), false));
+
+		Assert.assertEquals(dropdownItems.toString(), 2, dropdownItems.size());
+
+		DropdownItem dropdownItem1 = dropdownItems.get(0);
+
+		Assert.assertEquals("view", dropdownItem1.get("label"));
+
+		DropdownItem dropdownItem2 = dropdownItems.get(1);
+
+		Assert.assertEquals("delete", dropdownItem2.get("label"));
+
+		_resourcePermissionLocalService.removeResourcePermission(
+			TestPropsValues.getCompanyId(), objectDefinition.getClassName(),
+			ResourceConstants.SCOPE_COMPANY,
+			String.valueOf(TestPropsValues.getCompanyId()), role.getRoleId(),
+			ActionKeys.UPDATE);
+
+		dropdownItems = _fdsActionProvider.getDropdownItems(
+			TestPropsValues.getGroupId(),
+			_getMockHttpServletRequest(objectEntry.getObjectEntryId()),
+			new RelatedModel(
+				objectDefinition.getClassName(), objectEntry.getObjectEntryId(),
+				objectEntry.getTitleValue(), false));
+
+		Assert.assertEquals(dropdownItems.toString(), 1, dropdownItems.size());
+
+		dropdownItem1 = dropdownItems.get(0);
+
+		Assert.assertEquals("view", dropdownItem1.get("label"));
+	}
+
+	private MockHttpServletRequest _getMockHttpServletRequest(
+			long objectEntryId)
+		throws Exception {
+
+		MockHttpServletRequest mockHttpServletRequest =
+			new MockHttpServletRequest();
+
+		mockHttpServletRequest.setAttribute(
+			WebKeys.CURRENT_URL, "http://localhost:8080/currentURL");
+
+		ThemeDisplay themeDisplay = new ThemeDisplay();
+
+		themeDisplay.setCompany(
+			_companyLocalService.fetchCompany(TestPropsValues.getCompanyId()));
+		themeDisplay.setPermissionChecker(
+			PermissionThreadLocal.getPermissionChecker());
+		themeDisplay.setScopeGroupId(TestPropsValues.getGroupId());
+		themeDisplay.setUser(TestPropsValues.getUser());
+
+		mockHttpServletRequest.setAttribute(
+			WebKeys.THEME_DISPLAY, themeDisplay);
+
+		mockHttpServletRequest.addParameter(
+			"objectEntryId", String.valueOf(objectEntryId));
```

Comment:
I think we need to reorder because the way it is not ordered:
```
mockHttpServletRequest.setAttribute(
			WebKeys.CURRENT_URL, "http://localhost:8080/currentURL");
mockHttpServletRequest.setAttribute(
			WebKeys.THEME_DISPLAY, themeDisplay);
mockHttpServletRequest.addParameter(
			"objectEntryId", String.valueOf(objectEntryId));
```

If you were using `mockHttpServletRequest.setParameter()`, the order would be okay.

Reply:
Nice catch! Done [here](https://github.com/brianchandotcom/liferay-portal/pull/158416/commits/caf38042d94914af85a576d0aecbbe8b0ddff8c6)